### PR TITLE
Parse registered models

### PR DESCRIPTION
**Pull request by tonyelhabr**
Output looks like this

```
all_models <- list_registered_models(parse = TRUE)
> all_models
$registered_models
# A tibble: 7 × 4
  name                                     creation_timestamp  last_updated_timestamp latest_versions  
  <chr>                                    <dttm>              <dttm>                 <list>           
1 integration-test-22-01-20T02:12:13.13.81 2022-01-20 02:12:41 2022-01-20 02:12:41    <tibble [0 × 0]> 
2 integration-test-22-01-20T02:58:49.49.47 2022-01-20 02:58:52 2022-01-20 02:58:52    <tibble [0 × 0]> 
3 integration-test-22-01-20T03:00:33.33.97 2022-01-20 03:00:34 2022-01-20 03:00:34    <tibble [0 × 0]> 
4 integration-test-22-01-20T03:01:05.05.45 2022-01-20 03:01:06 2022-01-20 03:01:07    <tibble [1 × 10]>
5 integration-test-22-01-20T03:01:12.12.08 2022-01-20 03:01:13 2022-01-20 03:01:13    <tibble [0 × 0]> 
6 integration-test-22-01-20T03:02:18.18.67 2022-01-20 03:02:19 2022-01-20 03:02:30    <tibble [1 × 10]>
7 integration-test-22-01-20T03:02:41.41.12 2022-01-20 03:02:42 2022-01-20 03:02:42    <tibble [0 × 0]> 

$next_page_token
NULL
```

### CODE REVIEW

File: R/registry.R [hunk 1]
```diff
@@ -189,12 +191,53 @@ list_registered_models <- function(max_results = 100, page_token = NULL, client
...
+      run_link = map_chr(v, "run_link")
+    )
+  }
+  parsed_versions <- versions %>% map(parse_versions)
+
+  cbind(
+    parent,
+    tibble(latest_versions = parsed_versions)
+  ) %>%
+    as_tibble()
```
Comment (tonyelhabr) [inline R/registry.R:240]: avoid using `dplyr::bind_cols()` to avoid needing to depend on `dplyr`

File: R/registry.R [hunk 2]
```diff
@@ -189,12 +191,53 @@ list_registered_models <- function(max_results = 100, page_token = NULL, client
...
+  )
+  if(isFALSE(parse)) {
+    return(res)
+  }
+  res$registered_models <- parse_registered_models(res$registered_models)
+  res
+}
+
+#' @importFrom purrr transpose
+parse_registered_models <- function(registered_models) {
```
Comment (tonyelhabr) [inline R/registry.R:206]: not sure if it's worth exporting. i think we should if we don't end up going with the `parse` param

File: R/registry.R [hunk 3]
```diff
@@ -172,8 +172,10 @@ delete_registered_model <- function(name, client = mlflow_client()) {
 #' @param page_token Pagination token to go to the next page based on a
 #'   previous query.
 #' @param client An MLFlow client. Will be auto-generated if omitted.
+#' @param parse Logical indicating whether to return the `registered_models` element
+#' as a list (FALSE) or tibble (TRUE)
 #' @export
-list_registered_models <- function(max_results = 100, page_token = NULL, client = mlflow_client()) {
+list_registered_models <- function(max_results = 100, page_token = NULL, client = mlflow_client(), parse = FALSE) {
```
Comment (tonyelhabr) [inline R/registry.R:178]: arguably could leave this function as is and just force the user to call `parse()`

**Comment by mrkaye97** [comment]
alright so just to clarify: the idea here is to list _all_ registered models? what happens if one model has multiple versions? the `latest_versions` tibble-column will have multiple rows, and then we pluck out the registered model we want and then subset that tibble down to just the `Production` one?

**Comment by tonyelhabr** [comment]
> alright so just to clarify: the idea here is to list _all_ registered models? what happens if one model has multiple versions? the `latest_versions` tibble-column will have multiple rows, and then we pluck out the registered model we want and then subset that tibble down to just the `Production` one?

yes exactly (on that second part). multiple rows in the tibble-column if there are multiple registered models. 0 rows if there is none

**Review by mrkaye97 — APPROVED** [review]

**Pull request merged**